google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleFrameworkDetector.java

```diff
@@ -76,32 +69,39 @@ public FileType getFileType() {
   @NotNull
   @Override
   public ElementPattern<FileContent> createSuitableFilePattern() {
-    return new AppEngineFlexPattern().withAppEngineFlexYAMLContent();
+    return new AppEngineFlexPattern().withAppEngineFlexYamlContent();
   }
 
   /**
    * IntelliJ API pattern class that checks for App Engine Flex project file presence and checks it
    * has required configuration lines to avoid spurious detection.
    */
-  private static class AppEngineFlexPattern
+  private static final class AppEngineFlexPattern
       extends ObjectPattern<FileContent, AppEngineFlexPattern> {
 
     private AppEngineFlexPattern() {
       super(FileContent.class);
     }
 
-    private AppEngineFlexPattern withAppEngineFlexYAMLContent() {
+    private AppEngineFlexPattern withAppEngineFlexYamlContent() {
       return with(
           new PatternCondition<FileContent>("with-appengine-java-flexible") {
+            private final Collection<String> APP_ENGINE_FLEX_CONFIG_FILES =
```

Comment:
Whoops, sorry for the confusion, I meant to say put these constants in the `AppEngineFlexPattern` class. I've never seen these declared in an anonymous class before and I'm not sure how they will be handled (i.e. are they re-initialized for every instance of this anonymous class?).

Reply:
Agree, this is probably a bit too deep and plus it makes sense to make it static I think.